source/common/http/utility.cc

```diff
@@ -134,11 +134,13 @@ std::string Utility::parseCookieValue(const HeaderMap& headers, const std::strin
 }
 
 std::string Utility::makeSetCookieValue(const std::string& key, const std::string& value,
-                                        const std::chrono::seconds max_age) {
-  if (max_age == std::chrono::seconds::zero()) {
-    return fmt::format("{}=\"{}\"", key, value);
-  }
-  return fmt::format("{}=\"{}\"; Max-Age={}", key, value, max_age.count());
+                                        const std::string& path, const std::chrono::seconds max_age,
+                                        bool httponly) {
+  return fmt::format("{}=\"{}\"{}{}{}{}", key, value,
```

Comment:
This is starting to get pretty convoluted, to the point where it's difficult to understand what the output would actually look like. Can you refactor to get rid of the nested `fmt::format`s?

Reply:
Done, though I'll want a !Googler's take on using absl over fmt here.  Worst case we can do a bunch of string.appends

Oh, I also meant to call out the fact we're doing name="value" but path=/foo.  I'm inclined to take the quotes out around value but wanted to get your thoughts there - if you prefer them in both places or prefer them as-is that's fine too.

Reply:
Thanks, that looks a lot better with the StrAppend, but I confess I'm biased.

The rationale for quoting the value is that it gives the caller more flexibility. The HTTP spec disallows backslashes and double quotes in values to begin with, but this allows the caller to set a cookie that includes whitespace. Looking [at the Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) I don't think quoting the path is allowed.